follow_sponsorship/model/contracts.py

```diff
@@ -0,0 +1,277 @@
+        self._ins_wkf_items(
+            cr, uid, 'act_active',
+            wkf_id, active_contract_ids, context)
+
+    def _ins_wkf_items(self, cr, uid, act_id, wkf_id, cont_ids, context=None):
+            ir_model_data = self.pool.get('ir.model.data')
+            wkf_activity_id = ir_model_data.get_object_reference(
+                cr, uid, 'follow_sponsorship',
+                act_id)[1]
+
+            wkf_instance_ids = list()
+            for contract_id in cont_ids:
+                cr.execute(
+                    '''
+                    SELECT id FROM wkf_instance
+                    WHERE wkf_id = {} AND res_id = {}
+                    '''.format(wkf_id, contract_id))
+                wkf_instance_ids.append(cr.fetchall()[0][0])
+
+            for wkf_instance_id in wkf_instance_ids:
+                cr.execute(
+                    '''
+                    INSERT INTO wkf_workitem(act_id, inst_id)
+                    VALUES ('{}', '{}')
+                    '''.format(wkf_activity_id, wkf_instance_id)
+                )
+
+    # Only at module installation
+    def _set_sds_state(self, cr, uid, ids=None, context=None):
+        active_contract_ids = self.search(
+            cr, uid,
+            [('state', 'in', ['waiting', 'active', 'mandate'])],
+            context)
+        draft_contract_ids = self.search(
+            cr, uid,
+            [('state', '=', 'draft')],
+            context)
+
+        for active_contract_id in active_contract_ids:
+            cr.execute(
+                '''
+                UPDATE recurring_contract
+                SET sds_state = 'active'
+                WHERE id = '{}'
+                '''.format(active_contract_id)
+                )
+
+        for draft_contract_id in draft_contract_ids:
+            cr.execute(
+                '''
+                UPDATE recurring_contract
+                SET sds_state = 'draft'
+                WHERE id = '{}'
+                '''.format(draft_contract_id)
+                )
+
+    _columns = {
+        'sds_state': fields.selection([
+            ('draft', _('Draft')),
+            ('start', _('Start')),
+            ('waiting_welcome', _('Waiting welcome')),
+            ('active', _('Active')),
+            ('project_suspended', _('Project suspended')),
+            ('field_memo', _('Field memo')),
+            ('cancelled', _('Cancelled')),
+            ('sub_waiting', _('Sub waiting')),
+            ('no_sub', _('No sub')),
+            ('sub', _('Sub')),
+            ('sub_accept', _('Accept sub')),
+            ('sub_reject', _('Reject sub'))], _('SDS Status'), select=True,
+            readonly=True, track_visibility='onchange',
+            help=_('')),
+        'date_welcome': fields.date(_('Welcome date'), readonly=True),
+        'date_sub': fields.related(
+            'parent_id', 'start_date',
+            type='date', string=_('Date sub'),
+            store=False, readonly=True),

```

Comment:
You can store the field so that we can search and order sponsorships according to this field. You can also add this field in tree view and form view of contracts.